devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy

```diff
@@ -0,0 +1,346 @@
+	commands << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 0x05) // humidity
+
+	sendCommands(commands,1000)
+}
+
+/**
+ * Mapping of command classes and associated versions used for this DTH
+ */
+private getCommandClassVersions() {
+	[
+		0x20: 1,  // Basic
+		0x25: 1,  // Switch Binary
+		0x30: 1,  // Sensor Binary
+		0x31: 5,  // Sensor MultiLevel
+		0x56: 1,  // Crc16Encap
+		0x60: 3,  // Multi-Channel
+		0x70: 2,  // Configuration
+		0x98: 1,  // Security
+		0x71: 3   // Notification
+	]
+}
+
+def parse(String description) {
+	log.info("parse called: description: ${description}")
+	def result = []
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		result = zwaveEvent(cmd)
+	}
+	return createEvent(result)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd, endpoint=null) {
+	log.info "zwaveEvent BasicReport called: "+cmd +endpoint
+	def value = cmd.value ? "on" : "off"
+	endpoint ? changeSwitch(endpoint, value) : []
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd, endpoint = null) {
+	log.info "zwaveEvent SwitchBinaryReport called: ${endpoint}, ${cmd.value}"
+	def value = cmd.value ? "on" : "off"
+	endpoint ? changeSwitch(endpoint, value) : []
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd, endpoint = null) {
+	log.info "zwaveEvent SecurityMessageEncapsulation called: ${cmd}, ${endpoint}"
+	def encapsulatedCommand = cmd.encapsulatedCommand(commandClassVersions)
+	if (encapsulatedCommand) {
+		waveEvent(encapsulatedCommand)
+	} else {
+		log.warn "Unable to extract encapsulatedCommand from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd, endpoint = null) {
+	log.info "zwaveEvent MultiChannelCmdEncap called: '${cmd}'  endpoint '${endpoint}'"
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	log.debug "MultiChannelCmdEncap: encapsulatedCommand '${encapsulatedCommand}'"
+	zwaveEvent(encapsulatedCommand, cmd.sourceEndPoint as Integer)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd, endpoint = null) {
+	log.info "zwaveEvent NotificationReport called: notificationType: '${cmd.notificationType}', event: '${cmd.event}', endpoint '${endpoint}'"
+	def result = []
+
+	if (cmd.notificationType == 0x08) {
+		def value = cmd.event== 0x03? "on" : "off"
+		endpoint ? result = changeSwitch(endpoint, value) : []
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.crc16encapv1.Crc16Encap cmd) {
+	log.info "zwaveEvent Crc16Encap called: cmd '${cmd}'"
+	def versions = commandClassVersions
+	def version = versions[cmd.commandClass as Integer]
+	def ccObj = version ? zwave.commandClass(cmd.commandClass, version) : zwave.commandClass(cmd.commandClass)
+	def encapsulatedCommand = ccObj?.command(cmd.command)?.parse(cmd.data)
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	}
+	[:]
```

Comment:
This will overwrite the return value of the unencapsulated command handler.

Reply:
Ok. I will remove this line.